### DISCUSSION TITLE
Large Model Failures / Max size limit?

### DISCUSSION THREAD

**Discussion by andres-echo** [google/model-viewer#4925] [Q&A]
I am trying to use model viewer to render large GLBs (1GB ~ 4GB) within a browser javascript application. GLBs smaller than 2GB load fine but anything larger does not. I can confirm the device has more than enough memory (desktop PC) and the network tab shows the file being streamed and downloaded successfully (I am on ethernet so even a 4GB file streams in about a minute). There are no errors in the console, the progress bar for the viewer loads and fills but once done, the model does not. Is there a known limit to the size of GLBs that can be rendered within a browser? (Latest Chrome, W11). 

Thanks in advance!

**Comment by elalish** [comment]
Wow, I'm impressed the browser will render a 2GB file for you! Might be a GPU RAM limit, but more likely a browser JS limit. I'm sure there is one, though I don't know what it is - looks like you found it. 

Are you using KTX2 textures yet? That can help keep the memory footprint down. Beyond that, you may want to look at what e.g. Cesium does for breaking large scenes into LoD tiles and loading and unloading them dynamically as you move around. That's a bit beyond what MV is designed for.

**Reply by andres-echo** [reply]
Thanks so much for the quick reply!

Since I posted my question I found [this thread](https://issues.chromium.org/issues/328100615) which indicates a 4GB limit on js heap size in chromium browsers. I am guessing these models, combined with the page itself, causes the tab to hit the limit and error out (though I would have expected a little more noise from chrome about it; it fails very quietly!). 

Unfortunately the models I am trying to render are all from a client repository and fairly 'raw' (they are huge and I have no control over the textures therein). They are all discrete models (no scenes, movement etc just rendering individual models made from scans). It looks like my options are to compress the model before trying to render, or just don't render them😅

**Reply by elalish** [reply]
Yeah, +1 to compression. There's no way you're actually seeing 2GB worth of details on your screen anyway unless you zoom way in.
